src/schematools/utils.py

```diff
@@ -59,15 +67,22 @@ def profile_defs_from_url(profiles_url: Union[URL, str]) -> Dict[str, types.Prof
 
     The URL could be ``https://schemas.data.amsterdam.nl/profiles/``
     """
-    return defs_from_url(base_url=profiles_url, data_type=types.ProfileSchema)
+    return {
+        p_name: cast(types.ProfileSchema, p_schema)
+        for p_name, p_schema in defs_from_url(
+            base_url=profiles_url, data_type=types.ProfileSchema
+        ).items()
+    }
 
 
-def defs_from_url(base_url: Union[URL, str], data_type: Type[types.ST]) -> Dict[str, types.ST]:
+def defs_from_url(
+    base_url: Union[URL, str], data_type: Type[types.SchemaType]
+) -> Dict[str, types.SchemaType]:
```

Comment:
You should revert this change. By returning `SchemaType` here you need all those explicit upcasts in `schema_def_from_url` and `schema_defs_from_url`. With the original `TypeVar` definition `ST` those upcasts (from `SchemaType` to more specific type`) will not be necessary anymore.

Reply:
Fixed!